javav2/example_code/iotsitewise/src/main/java/com/example/iotsitewise/scenario/SitewiseActions.java

```diff
@@ -0,0 +1,479 @@
+        ListAssetModelPropertiesRequest modelPropertiesRequest = ListAssetModelPropertiesRequest.builder().assetModelId(assetModelId).build();
+        return getAsyncClient().listAssetModelProperties(modelPropertiesRequest)
+                .handle( (response, throwable) -> {
+                    if (response != null){
+                        return  response.assetModelPropertySummaries().stream()
+                                .collect(Collectors
+                                        .toMap(AssetModelPropertySummary::name, AssetModelPropertySummary::id));
+                    } else {
+                        throw (CompletionException) throwable.getCause();
+                    }
+                });
+    }
+    // snippet-end:[sitewise.java2.describe.asset.model.main]
+
+    // snippet-start:[sitewise.java2.delete.asset.main]
+    /**
+     * Deletes an asset asynchronously.
+     *
+     * @param assetId the ID of the asset to be deleted
+     * @return a {@link CompletableFuture} that represents the asynchronous operation of deleting the asset
+     * @throws RuntimeException if the asset deletion fails
+     */
+    public CompletableFuture<DeleteAssetResponse> deleteAssetAsync(String assetId) {
+        DeleteAssetRequest deleteAssetRequest = DeleteAssetRequest.builder()
+            .assetId(assetId)
+            .build();
+
+        return getAsyncClient().deleteAsset(deleteAssetRequest)
+            .whenComplete((response, exception) -> {
+                if (exception != null) {
+                    logger.error("An error occurred deleting asset with id: {}", assetId);
+                } else {
+                    logger.info("Request to delete the asset completed successfully.");
+                }
+            });
+    }
+    // snippet-end:[sitewise.java2.delete.asset.main]
+
+    // snippet-start:[sitewise.java2.delete.asset.model.main]
+    /**
+     * Asynchronously deletes an Asset Model with the specified ID.
+     *
+     * @param assetModelId the ID of the Asset Model to delete
+     * @return a {@link CompletableFuture} that completes with the {@link DeleteAssetModelResponse} when the operation is complete
+     * @throws RuntimeException if the operation fails, containing the error message and the underlying exception
+     */
+    public CompletableFuture<DeleteAssetModelResponse> deleteAssetModelAsync(String assetModelId) {
+        DeleteAssetModelRequest deleteAssetModelRequest = DeleteAssetModelRequest.builder()
+            .assetModelId(assetModelId)
+            .build();
+
+        return getAsyncClient().deleteAssetModel(deleteAssetModelRequest)
+            .whenComplete((response, exception) -> {
+                if (exception != null) {
+                    throw new RuntimeException("Failed to delete asset model with ID: " + assetModelId + ". Error: " + exception.getMessage(), exception);
+                }
+            });
+    }
```

Comment:
I still think we need to review what the CompletableFuture should do with an exception. My opinion is that at most it should just log the exception. Here you don't need to rethrow it, since the join method will throw it anyway.

Reply:
Lets discuss and update the standards.

Reply:
The code team wants exceptions bubbled up to calling code